src/app/auth/login/login.component.ts

```diff
@@ -1,74 +1,71 @@
-import { AuthService } from '../../guard/auth.service';
-import { AppState, User } from '../../app.state';
-import { forbiddenNameValidator } from '../../shared/forbidden-name-validator.directive';
-import { AuthActions } from '../action-types';
-import { isLoggedIn } from '../auth.selectors';
+import {Store} from '@ngrx/store';
+import {AuthService} from '../../guard/auth.service';
+import {AppState} from '../../app.state';
+import {AuthActions} from '../action-types';
+import {getAuthError, isLoggedIn} from '../auth.selectors';
+import {LoaderComponent} from '../../loader/loader.component';
 
 @Component({
   selector: 'app-login',
   standalone: true,
   templateUrl: './login.component.html',
   styleUrl: './login.component.scss',
-  imports: [CommonModule, RouterModule, ReactiveFormsModule],
+  imports: [CommonModule, RouterModule, ReactiveFormsModule, NgOptimizedImage, LoaderComponent],
 })
-export class LoginComponent implements OnInit {
+export class LoginComponent implements OnInit, OnDestroy {
   public signupForm!: FormGroup;
-  public isLoading = false;
-  public errorMessage = '';
-  isLoggenIn!: Signal<boolean>;
-  constructor(
+  public errorMessage = this.store.selectSignal(getAuthError);
+  public isLoggedIn = this.store.selectSignal(isLoggedIn);
+  private authSubscription!: Subscription;
+
+  public constructor(
+    private formBuilder: FormBuilder,
     private authService: AuthService,
     private router: Router,
     private store: Store<AppState>
-  ) {}
+  ) {
+  }
 
-  ngOnInit(): void {
-    this.signupForm = new FormGroup({
-      email: new FormControl(null, Validators.required),
-      password: new FormControl(null, Validators.required),
+  public ngOnInit(): void {
+    this.signupForm = this.formBuilder.group({
+      email: ['', [Validators.required, Validators.email]],
+      password: ['', [Validators.required, Validators.minLength(8)]],
     });
-    this.isLoggenIn = this.store.selectSignal(isLoggedIn);
-    if (this.isLoggenIn()) {
-      this.router.navigate(['/home']);
+    if (this.isLoggedIn()) {
+      this.router.navigateByUrl('/home');
     }
   }
 
   public onSubmit() {
     const email = this.signupForm.value.email;
     const password = this.signupForm.value.password;
-
-    let authObs: Observable<User>;
-    this.isLoading = true;
-
-    authObs = this.authService.logIn(email, password);
-
-    authObs
+    const authObs = this.authService.logIn(email, password);
+    this.authSubscription = authObs
       .pipe(
         tap((user) => {
-          this.store.dispatch(AuthActions.login({ user: { ...user } }));
-          this.router.navigate(['/home']);
+          this.store.dispatch(AuthActions.login({user: {...user, isLoading: false, error: null}}));
+          this.router.navigateByUrl('/home');
+        }),
+        catchError((error) => {
+          this.store.dispatch(AuthActions.loginError({error: error.message || 'Login failed'}));
+          return of();
```

Comment:
what is this doing here? the return of().

And again I feel you are miss using the tap operator